extensions/whatsapp/src/utils/get-apple-contacts.ts

```diff
@@ -0,0 +1,36 @@
+import { AppleContact, isAppleContact } from "./types";
+import { runAppleScript } from "run-applescript";
+import fs from "fs/promises";
+import os from "os";
+import path from "path";
+import { randomId } from "@raycast/api";
+
+export async function getAppleContacts(): Promise<Array<AppleContact>> {
+  const fileName = randomId();
+
+  await runAppleScript(`
+    tell application "Contacts"
+      set the clipboard to (vcard of people) as text
+      do shell script "LANG=en_US.UTF-8 pbpaste >~/Desktop/${fileName}.vcf"
+      set the clipboard to ""
+    end tell`
+  );
+
+  const filePath = path.join(os.homedir(), "Desktop", `${fileName}.vcf`);
```

Comment:
Why do you need to write to a file?

Reply:
Same question, does it really needed here?

Reply:
After some trial and error I did not find a way to get the contact list without it being slow. Maybe you know a better way since you have a native `Search Contacts` command.